extensions/GitExtension/FileExplorerGitIntegration/Models/GitLocalRepository.cs

```diff
@@ -0,0 +1,230 @@
+                case "System.VersionControl.LastChangeAuthorEmail":
+                    if (latestCommit == null)
+                    {
+                        latestCommit = FindLatestCommit(relativePath, repository);
+                        if (latestCommit != null)
+                        {
+                            result.Add("System.VersionControl.LastChangeAuthorEmail", latestCommit.Author.Email);
+                        }
+                    }
+                    else
+                    {
+                        result.Add("System.VersionControl.LastChangeAuthorEmail", latestCommit.Author.Email);
+                    }
+
+                    break;
+                case "System.VersionControl.LastChangeID":
+                    if (latestCommit == null)
+                    {
+                        latestCommit = FindLatestCommit(relativePath, repository);
+                        if (latestCommit != null)
+                        {
+                            result.Add("System.VersionControl.LastChangeID", latestCommit.Sha);
+                        }
+                    }
+                    else
+                    {
+                        result.Add("System.VersionControl.LastChangeID", latestCommit.Sha);
+                    }
+
+                    break;
+                case "System.VersionControl.Status":
+                    result.Add("System.VersionControl.Status", GetStatus(relativePath, repository));
+                    break;
+
+                case "System.VersionControl.CurrentFolderStatus":
+                    var folderStatus = repository.Head.FriendlyName + " AheadBy: " + repository.Head.TrackingDetails.AheadBy + " BehindBy: " + repository.Head.TrackingDetails.BehindBy;
+                    result.Add("System.VersionControl.CurrentFolderStatus", folderStatus);
+                    break;
+            }
+        }
+
+        log.Debug("Returning source control properties from git source control extension");
+        return result;
+    }
+
+    public IPropertySet GetProperties(string[] properties, string relativePath)
+    {
+        return ((ILocalRepository)this).GetProperties(properties, relativePath);
+    }
+
+    private string? GetStatus(string relativePath, Repository repository)
+    {
+        try
+        {
+            var status = repository.RetrieveStatus(relativePath);
+            if (status == FileStatus.Unaltered)
+            {
+                return string.Empty;
+            }
+
+            return status.ToString();
+        }
+        catch
+        {
+            return null;
+        }
+    }
+
+    private Commit? FindLatestCommit(string relativePath, Repository repository)
+    {
+        var commits = repository.Commits;
+
+        // Check the most recent commit and bail if the file is not present
+        {
+            var firstCommit = commits.FirstOrDefault();
+            if (firstCommit != null && firstCommit.Tree[relativePath] == null)

```

Comment:
I believe this can be simplified to commits.FirstOrDefault()?.Tree?[relativePath] == null